cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -194,6 +197,19 @@ public KubeTwillRunnerService(MasterEnvironmentContext masterEnvContext, ApiClie
     this.workloadIdentityProvider = workloadIdentityProvider;
   }
 
+  private String generateSelector() {
```

Comment:
Do you try to have two different preview sets for manager and appfabric? If so, why? Why does app fabric need to manage this stateful set

Reply:
No, the plan is for AppFabric to not manage it. Only Preview manager should.

Reply:
Then why selector needs to know the current pod and such?

Reply:
Can we have only app-fabric to manage all?

Reply:
I will refactor this PR to better provide the selector with information as to what to watch out for. So that the information is coming from the Pods. Used this PR to get comments to confirm functionality.

Will remove the hard-coded values from KubeTwillRunnerService.

Reply:
> Can we have only app-fabric to manage all?

The preview-runners are created by the preview pod. So the separation seems cleaner if we let preview manager pod watch these resources. And app fabric watches the resources it created.

I am modifying this PR, so that the hacky approach in this first PR is replaced.

Reply:
> Then why selector needs to know the current pod and such?

I have modified the commit so that the approach is generic. 
In this PR, there is a parent.pod label. 

As we discussed offline, it would be worth it to explore if we can have a service name instead of pod name. So that if we try for replication of app-fabric or preview pods in the future, it would be good. 

I am looking at possible ways to do so.